internal/internal_workflow_client.go

```diff
@@ -1800,7 +1800,7 @@ func (w *workflowClientInterceptor) UpdateWithStartWorkflow(
 	}
 	handle, err := w.updateHandleFromResponse(ctx, updateReq.WaitPolicy.LifecycleStage, updateResp)
 	if err != nil {
-		return nil, fmt.Errorf("%w: %w", errInvalidWithStartWorkflowOperation, err)
+		return nil, err
```

Comment:
Wrapping with `errInvalidWithStartWorkflowOperation` ("invalid WithStartWorkflowOperation") is inappropriate IMO, since it's not related to the start operation.